morf-sqlserver/src/main/java/org/alfasoftware/morf/jdbc/sqlserver/SqlServerDialect.java

```diff
@@ -479,7 +506,28 @@ protected String getSqlFrom(UpdateStatement statement) {
     return sqlBuilder.toString();
   }
 
+
   /**
+   * @see SqlServerDialect#getSqlFrom(SequenceReference)
+   */
+  @Override
+    protected String getSqlFrom(SequenceReference sequenceReference) {
+    StringBuilder result = new StringBuilder();
+
+    switch (sequenceReference.getTypeOfOperation()) {
+      case NEXT_VALUE:
+        result.append("NEXT VALUE FOR ");
+        break;
+      case CURRENT_VALUE:
+        return "current_value FROM sys.sequences WHERE name = '" + sequenceReference.getName() + "'";

```

Comment:
I think this is wrong.
It has to be a full sub-select, including the brackets to make it a field-value again:
```
return "(SELECT current_value FROM sys.sequences WHERE name = '" + sequenceReference.getName() + "')";
````